### ISSUE TITLE
Bad gt line error

### ISSUE THREAD

**Issue by victorygod** [MichalBusta/DeepTextSpotter#39]
I am trying to run train.py with my own data. However, I got the bad gt line error for every line when the program reading my ground truth label files. 

The format in my file is like this:
x, y, w, h, words
x2, y2, w2, h2, words

I changed the value of data_param.scource in tiny.prototxt to my own path. 

I couldn't find where do you print the bad gt line error in python code, so it is hard for me to debug. Could you indicate where is the data loading process for train.py? Or could you make it modifiable in python part? Because I really don't know where to locate the error. 

Thanks.

**Comment by MichalBusta** [comment]
https://github.com/MichalBusta/DeepTextSpotter/issues/10

**Comment by victorygod** [comment]
Thanks for your advice!

I use the CIDAR-focused dataset. Now my data look like this:

0 0.5203125 0.21458333333333332 0.5325 0.135 0 PROPER 
0 0.5234375 0.5104166666666666 0.535 0.205 0 FOOD 
0 0.521875 0.7947916666666667 0.53 0.13125 0 PRONTO 

I think the data should be good this time.

I use the pre-trained network you provided. However, The OCR network just couldn't output any text. Here is a snapshot:

![loss](https://user-images.githubusercontent.com/7223522/41029684-e4f670ca-69ae-11e8-8457-2d3c8b570db2.png)

I output the scaled images which should be the input of the OCR network, it seems very well:

![scaled](https://user-images.githubusercontent.com/7223522/41029850-665d68f8-69af-11e8-8668-d56b005bf1c1.jpg)

I am not sure why the OCR network couldn't output anything. I also tried to change the input image from RGB to BGR. The result remains the same. 

I tried to output the transpose layer of the OCR and I found that it responses the same value for different inputs. So I think there must be some layer in the OCR network that always output 0.

Could give me some more advice on some possible reason on why all these happened?

**Comment by MichalBusta** [comment]
On 06/06/2018 11:39 AM, Hantian Li wrote:
>
> Thanks for your advice!
>
> I use the CIDAR-focused dataset. Now my data look like this:
>
> 0 0.5203125 0.21458333333333332 0.5325 0.135 0 PROPER
> 0 0.5234375 0.5104166666666666 0.535 0.205 0 FOOD
> 0 0.521875 0.7947916666666667 0.53 0.13125 0 PRONTO
>
> I think the data should be good this time.
>
> I use the pre-trained network you provided. However, The OCR network 
> just couldn't output any text. Here is a snapshot:
>
> loss 
> <https://user-images.githubusercontent.com/7223522/41029684-e4f670ca-69ae-11e8-8457-2d3c8b570db2.png>
>
> I output the scaled images which should be the input of the OCR 
> network, it seems very well:
>
> scaled 
> <https://user-images.githubusercontent.com/7223522/41029850-665d68f8-69af-11e8-8668-d56b005bf1c1.jpg>
>
> I am not sure why the OCR network couldn't output anything. I also 
> tried to change the input image from RGB to BGR. The result remains 
> the same.
>
ocr is gray-scale
>
> I tried to output the transpose layer of the OCR and I found that it 
> responses the same value for different inputs. So I think there must 
> be some layer in the OCR network that always output 0.
>
> Could give me some more advice on some possible reason on why all 
> these happened?
>
Just quick tips:
  - ctc loss value shoud be around 1.1 - you are about > 20 - so most 
probable label is 0
  - isolate problem - just run OCR on patches
  - there is debug option in python script - it can help
